src/analysis/refactor_extract_region.ml

```diff
@@ -0,0 +1,395 @@
+open Std
+
+exception Nothing_to_do
+exception Not_allowed_in_interface_file
+
+let () =
+  Location.register_error_of_exn (function
+    | Nothing_to_do -> Some (Location.error "Nothing to do")
+    | Not_allowed_in_interface_file ->
+      Some
+        (Location.error
+           "Expression extraction is only allowed in implementation file")
+    | _ -> None)
```

Comment:
Iam not sure about `register_error`. Does the result is wrapped into an error like other commands?

Reply:
Yes, the error message looks like this:
```json
{
   "class": "error",
   "value": "Nothing to do",
   "notifications": []
}
```

Reply:
Perfect!